mock/storage_reads.go

```diff
@@ -78,7 +78,7 @@ type GroupResultSet struct {
 	ErrFunc   func() error
 }
 
-func NewGroupResultSet() *GroupResultSet {
+func NewGroupedResultSet() *GroupResultSet {
```

Comment:
I don't have a strong opinion either way, but it would be good for the name of this function to match the type it returns, either `Group` or `Grouped`.